gumroad/includes/scanner.php

```diff
@@ -24,7 +24,7 @@ function gum_post_scanner( $post_object ) {
  function scanner( $post_content ) {
 
  	// counts to see if we find any gumroad product links.
- 	if (count(preg_grep('/(gumroad.com\/l\/[[:alnum:]]|gum.co\/[[:alnum:]])/', explode("\n", $post_content))) > 0) {
+ 	if (count(preg_grep('/(gumroad.com\/a\/.+\/[[:alnum:]].*|gumroad.com\/l\/[[:alnum:]]|gum.co\/[[:alnum:]])/', explode("\n", $post_content))) > 0) {
```

Comment:
Also, let's use `[[:alnum:]]+` instead of `.+`? (affiliate ID is only numbers now, but let's keep the support for alphabets too)

```suggestion
 	if (count(preg_grep('/(gumroad.com\/a\/[[:alnum:]]+\/[[:alnum:]].*|gumroad.com\/l\/[[:alnum:]]|gum.co\/[[:alnum:]])/', explode("\n", $post_content))) > 0) {
```